backup: TestCloudBackupRestoreAzure failed [missing ResumingReader close call]

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21013350?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21013350?buildTab=artifacts#/) on master @ [bd57614e1f04a1ebfc1515d2fee80bf628eb4331](https://github.com/cockroachdb/cockroach/commits/bd57614e1f04a1ebfc1515d2fee80bf628eb4331):


```
        created by golang.org/x/net/http2.(*Transport).newClientConn in goroutine 17955
        	external/org_golang_x_net/http2/transport.go:866 +0xd85
        goroutine 52108 [select, "job":"1143790203354841089", "n":"1", "restore-span":"0", "restore-worker":"4"]:
        net/http.(*persistConn).writeLoop(0xc00e505680)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 52267
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 38206 [select, "job":"BACKUP id=1143790158276296705", "n":"1"]:
        net/http.(*persistConn).writeLoop(0xc0068466c0)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 38161
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 41918 [IO wait]:
        internal/poll.runtime_pollWait(0x7f32d896ae00, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc0020d8e80?, 0xc00952e000?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc0020d8e80, {0xc00952e000, 0xa000, 0xa000})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc0020d8e80, {0xc00952e000?, 0x83b7b1?, 0xc00952e005?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc005959a58, {0xc00952e000?, 0x0?, 0x9ffb?})
        	GOROOT/src/net/net.go:196 +0x45
        github.com/cockroachdb/cockroach/pkg/util/cidr.metricsConn.Read({{0x7c67f00?, 0xc005959a58?}, 0xc00142e740?, 0xc00142e750?}, {0xc00952e000?, 0x431174?, 0xc00952e005?})
        	pkg/util/cidr/cidr.go:519 +0x3e
        crypto/tls.(*atLeastReader).Read(0xc00c0355d8, {0xc00952e000?, 0x9f0a?, 0x4b871d?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc0091e29a8, {0x7bc3540, 0xc00c0355d8})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc0091e2708, {0x7f32d1bc8520, 0xc00c986c20}, 0x45b514?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc0091e2708, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc0091e2708, {0xc008a03000, 0x1000, 0xc00173fb00?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        net/http.(*persistConn).Read(0xc00b4c0120, {0xc008a03000?, 0x91acc5?, 0x7bb4cc0?})
        	GOROOT/src/net/http/transport.go:2125 +0x47
        bufio.(*Reader).fill(0xc0071ef500)
        	GOROOT/src/bufio/bufio.go:113 +0x103
        bufio.(*Reader).Peek(0xc0071ef500, 0x1)
        	GOROOT/src/bufio/bufio.go:152 +0x53
        net/http.(*persistConn).readLoop(0xc00b4c0120)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 41923
        	GOROOT/src/net/http/transport.go:1947 +0x174f
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_explicit/no_kms (50.37s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59007

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21038434?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21038434?buildTab=artifacts#/) on master @ [f0bbee90d39210a4085014d340fa432cb652d123](https://github.com/cockroachdb/cockroach/commits/f0bbee90d39210a4085014d340fa432cb652d123):


```
        created by net/http.(*Transport).dialConn in goroutine 44263
        	GOROOT/src/net/http/transport.go:1947 +0x174f
        goroutine 79210 [select]:
        net/http.(*persistConn).writeLoop(0xc006988000)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 79186
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 76088 [IO wait, "job":"BACKUP id=1145489540416765953", "n":"1"]:
        internal/poll.runtime_pollWait(0x7f1726d13400, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc00211f080?, 0xc0083ac000?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc00211f080, {0xc0083ac000, 0xa000, 0xa000})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc00211f080, {0xc0083ac000?, 0x83b8d1?, 0xc0083ac005?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc0098ab3a8, {0xc0083ac000?, 0x0?, 0xc009b27908?})
        	GOROOT/src/net/net.go:196 +0x45
        github.com/cockroachdb/cockroach/pkg/util/cidr.metricsConn.Read({{0x7ccc340?, 0xc0098ab3a8?}, 0xc00fed6620?, 0xc00fed6630?}, {0xc0083ac000?, 0x431174?, 0xc0083ac005?})
        	pkg/util/cidr/cidr.go:519 +0x3e
        crypto/tls.(*atLeastReader).Read(0xc010174ac8, {0xc0083ac000?, 0xc009b2790a?, 0x43ac36?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc006a6e628, {0x7c27900, 0xc010174ac8})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc006a6e388, {0x7f17295af7a8, 0xc00b095140}, 0x45b514?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc006a6e388, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc006a6e388, {0xc004789000, 0x1000, 0xc00e6ea7e0?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        net/http.(*persistConn).Read(0xc00b6486c0, {0xc004789000?, 0x91ade5?, 0x7c19000?})
        	GOROOT/src/net/http/transport.go:2125 +0x47
        bufio.(*Reader).fill(0xc00820d620)
        	GOROOT/src/bufio/bufio.go:113 +0x103
        bufio.(*Reader).Peek(0xc00820d620, 0x1)
        	GOROOT/src/bufio/bufio.go:152 +0x53
        net/http.(*persistConn).readLoop(0xc00b6486c0)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 76122
        	GOROOT/src/net/http/transport.go:1947 +0x174f
        goroutine 17177 [select, "job":"BACKUP id=1145489339883945985", "n":"1"]:
        net/http.(*persistConn).writeLoop(0xc008ca4900)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 17168
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_explicit/kms_auth_implicit (61.37s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21045198?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21045198?buildTab=artifacts#/) on master @ [6a6894a1d366cc177d25cb926a7b1434d3b56703](https://github.com/cockroachdb/cockroach/commits/6a6894a1d366cc177d25cb926a7b1434d3b56703):


```
        created by net/http.(*Transport).dialConn in goroutine 7275
        	GOROOT/src/net/http/transport.go:1947 +0x174f
        goroutine 7324 [select, "job":"1145772125814554625", "n":"1", "restore-span":"1", "restore-worker":"1"]:
        net/http.(*persistConn).writeLoop(0xc007324fc0)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 7378
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 6449 [select, "job":"1145772125814554625", "n":"1", "restore-span":"0", "restore-worker":"6"]:
        net/http.(*persistConn).writeLoop(0xc005f70120)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 7275
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 7323 [IO wait, "job":"1145772125814554625", "n":"1", "restore-span":"1", "restore-worker":"1"]:
        internal/poll.runtime_pollWait(0x7fa8c2e42600, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc006fb3200?, 0xc004390000?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc006fb3200, {0xc004390000, 0xa000, 0xa000})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc006fb3200, {0xc004390000?, 0x83b8d1?, 0xc004366005?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc00319e060, {0xc004390000?, 0x0?, 0xc005bb4908?})
        	GOROOT/src/net/net.go:196 +0x45
        github.com/cockroachdb/cockroach/pkg/util/cidr.metricsConn.Read({{0x7cee620?, 0xc00319e060?}, 0xc002914610?, 0xc002914630?}, {0xc004390000?, 0x431174?, 0xc004366005?})
        	pkg/util/cidr/cidr.go:519 +0x3e
        crypto/tls.(*atLeastReader).Read(0xc00523bb30, {0xc004390000?, 0xc005bb490a?, 0x43ac36?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc007d242a8, {0x7c49ae0, 0xc00523bb30})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc007d24008, {0x7fa8c212f7c0, 0xc0041bcb20}, 0x45b514?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc007d24008, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc007d24008, {0xc007efc000, 0x1000, 0xc00817f2c0?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        net/http.(*persistConn).Read(0xc007324fc0, {0xc007efc000?, 0x91ade5?, 0x7c3b1a0?})
        	GOROOT/src/net/http/transport.go:2125 +0x47
        bufio.(*Reader).fill(0xc00702fb60)
        	GOROOT/src/bufio/bufio.go:113 +0x103
        bufio.(*Reader).Peek(0xc00702fb60, 0x1)
        	GOROOT/src/bufio/bufio.go:152 +0x53
        net/http.(*persistConn).readLoop(0xc007324fc0)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 7378
        	GOROOT/src/net/http/transport.go:1947 +0x174f
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_legacy/no_kms (53.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21058135?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21058135?buildTab=artifacts#/) on master @ [0fbbeaf898276fbece74f6c5cb6615b8ee215400](https://github.com/cockroachdb/cockroach/commits/0fbbeaf898276fbece74f6c5cb6615b8ee215400):


```
        crypto/tls.(*Conn).Read(0xc0084a8388, {0xc0082f7000, 0x1000, 0xc005afc540?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        net/http.(*persistConn).Read(0xc00a07c480, {0xc0082f7000?, 0x91ade5?, 0x7c41ba0?})
        	GOROOT/src/net/http/transport.go:2125 +0x47
        bufio.(*Reader).fill(0xc00a0decc0)
        	GOROOT/src/bufio/bufio.go:113 +0x103
        bufio.(*Reader).Peek(0xc00a0decc0, 0x1)
        	GOROOT/src/bufio/bufio.go:152 +0x53
        net/http.(*persistConn).readLoop(0xc00a07c480)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 15955
        	GOROOT/src/net/http/transport.go:1947 +0x174f
        goroutine 23042 [IO wait, "job":"1146904231581614081", "n":"1", "restore-span":"0", "restore-worker":"0"]:
        internal/poll.runtime_pollWait(0x7f498634ee00, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc00221f600?, 0xc007e1c000?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc00221f600, {0xc007e1c000, 0x8000, 0x8000})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc00221f600, {0xc007e1c000?, 0x83b8d1?, 0xc007e1c005?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc007e18bd8, {0xc007e1c000?, 0x0?, 0xc00845e908?})
        	GOROOT/src/net/net.go:196 +0x45
        github.com/cockroachdb/cockroach/pkg/util/cidr.metricsConn.Read({{0x7cf5080?, 0xc007e18bd8?}, 0xc001f46af0?, 0xc001f46b00?}, {0xc007e1c000?, 0x431174?, 0x7f4989738290?})
        	pkg/util/cidr/cidr.go:519 +0x3e
        crypto/tls.(*atLeastReader).Read(0xc00713b2d8, {0xc007e1c000?, 0xc00845e90a?, 0x43ac36?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc00643ed28, {0x7c504e0, 0xc00713b2d8})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc00643ea88, {0x7f49863ef098, 0xc00946d080}, 0x45b514?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc00643ea88, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc00643ea88, {0xc00b95f000, 0x1000, 0xc00a488c90?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        net/http.(*persistConn).Read(0xc00918cc60, {0xc00b95f000?, 0x91ade5?, 0x7c41ba0?})
        	GOROOT/src/net/http/transport.go:2125 +0x47
        bufio.(*Reader).fill(0xc00986bb60)
        	GOROOT/src/bufio/bufio.go:113 +0x103
        bufio.(*Reader).Peek(0xc00986bb60, 0x1)
        	GOROOT/src/bufio/bufio.go:152 +0x53
        net/http.(*persistConn).readLoop(0xc00918cc60)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 23011
        	GOROOT/src/net/http/transport.go:1947 +0x174f
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_legacy/kms_auth_explicit (64.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21160939?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21160939?buildTab=artifacts#/) on master @ [cccaf85c81ec75541a93cd755013155961b1bb8a](https://github.com/cockroachdb/cockroach/commits/cccaf85c81ec75541a93cd755013155961b1bb8a):


```
        	external/org_golang_x_net/http2/frame.go:505 +0x6b
        golang.org/x/net/http2.(*Framer).ReadFrame(0xc001e262a0)
        	external/org_golang_x_net/http2/frame.go:564 +0x18
        golang.org/x/net/http2.(*clientConnReadLoop).run(0xc00a49cfa8)
        	external/org_golang_x_net/http2/transport.go:2208 +0xca
        golang.org/x/net/http2.(*ClientConn).readLoop(0xc007c8e700)
        	external/org_golang_x_net/http2/transport.go:2077 +0x52
        created by golang.org/x/net/http2.(*Transport).newClientConn in goroutine 23437
        	external/org_golang_x_net/http2/transport.go:866 +0xd85
        goroutine 23732 [select, "job":"BACKUP id=1153415472818159617", "n":"1"]:
        net/http.(*persistConn).writeLoop(0xc006b52120)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 23662
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 23731 [IO wait, "job":"BACKUP id=1153415472818159617", "n":"1"]:
        internal/poll.runtime_pollWait(0x7f550a86ea00, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc00b4a4a00?, 0xc0001e0500?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc00b4a4a00, {0xc0001e0500, 0x2500, 0x2500})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc00b4a4a00, {0xc0001e0500?, 0xc0001e0500?, 0x5?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc00afa88a8, {0xc0001e0500?, 0x7f550a961060?, 0x7f5555222f30?})
        	GOROOT/src/net/net.go:196 +0x45
        crypto/tls.(*atLeastReader).Read(0xc006f2a390, {0xc0001e0500?, 0x240a?, 0x43ac36?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc0057ef428, {0x7d6cf20, 0xc006f2a390})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc0057ef188, {0x7d6b9c0, 0xc00afa88a8}, 0x45b514?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc0057ef188, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc0057ef188, {0xc00786c000, 0x1000, 0xc00621b740?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        net/http.(*persistConn).Read(0xc006b52120, {0xc00786c000?, 0x91a8a5?, 0x7d5e480?})
        	GOROOT/src/net/http/transport.go:2125 +0x47
        bufio.(*Reader).fill(0xc007a0b920)
        	GOROOT/src/bufio/bufio.go:113 +0x103
        bufio.(*Reader).Peek(0xc007a0b920, 0x1)
        	GOROOT/src/bufio/bufio.go:152 +0x53
        net/http.(*persistConn).readLoop(0xc006b52120)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 23662
        	GOROOT/src/net/http/transport.go:1947 +0x174f
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_legacy/kms_auth_implicit (63.42s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21177352?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21177352?buildTab=artifacts#/) on master @ [0a2d99d3fc6dced4f627d643ddb65e790a7f9b7b](https://github.com/cockroachdb/cockroach/commits/0a2d99d3fc6dced4f627d643ddb65e790a7f9b7b):


```
        net/http.(*persistConn).readLoop(0xc00649bd40)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 84268
        	GOROOT/src/net/http/transport.go:1947 +0x174f
        goroutine 93262 [select]:
        net/http.(*persistConn).writeLoop(0xc00af25560)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 93244
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 21990 [select]:
        net/http.(*persistConn).writeLoop(0xc004b786c0)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 21891
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 14533 [IO wait, "job":"BACKUP id=1154264778853089281", "n":"sql1"]:
        internal/poll.runtime_pollWait(0x7f0cad85de00, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc0017d8980?, 0xc0079e4a00?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc0017d8980, {0xc0079e4a00, 0x2500, 0x2500})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc0017d8980, {0xc0079e4a00?, 0xc0079e4a00?, 0x5?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc002179a88, {0xc0079e4a00?, 0x7f0caf440ae0?, 0x7f0cfae65108?})
        	GOROOT/src/net/net.go:196 +0x45
        crypto/tls.(*atLeastReader).Read(0xc007ac2990, {0xc0079e4a00?, 0x240a?, 0x278?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc006d0bb28, {0x7d80be0, 0xc007ac2990})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc006d0b888, {0x7d7f840, 0xc002179a88}, 0x45b514?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc006d0b888, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc006d0b888, {0xc009f17000, 0x1000, 0xc0021b34d0?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        net/http.(*persistConn).Read(0xc009be9e60, {0xc009f17000?, 0x91a8a5?, 0x7d72120?})
        	GOROOT/src/net/http/transport.go:2125 +0x47
        bufio.(*Reader).fill(0xc00d9fa300)
        	GOROOT/src/bufio/bufio.go:113 +0x103
        bufio.(*Reader).Peek(0xc00d9fa300, 0x1)
        	GOROOT/src/bufio/bufio.go:152 +0x53
        net/http.(*persistConn).readLoop(0xc009be9e60)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 14514
        	GOROOT/src/net/http/transport.go:1947 +0x174f
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_implicit/kms_auth_explicit (59.52s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21213611?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21213611?buildTab=artifacts#/) on master @ [7a924823002246b4fb02b54c3e71639d7a7a1e7c](https://github.com/cockroachdb/cockroach/commits/7a924823002246b4fb02b54c3e71639d7a7a1e7c):


```
        goroutine 7264 [select, "job":"1156248981336588289", "n":"1", "restore-span":"0", "restore-worker":"3"]:
        net/http.(*persistConn).writeLoop(0xc00d31eb40)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 7279
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        goroutine 7263 [IO wait, "job":"1156248981336588289", "n":"1", "restore-span":"0", "restore-worker":"3"]:
        internal/poll.runtime_pollWait(0x7f8c53601e00, 0x72)
        	GOROOT/src/runtime/netpoll.go:351 +0x85
        internal/poll.(*pollDesc).wait(0xc010c25700?, 0xc00d6ec000?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x27
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc010c25700, {0xc00d6ec000, 0xa000, 0xa000})
        	GOROOT/src/internal/poll/fd_unix.go:165 +0x279
        net.(*netFD).Read(0xc010c25700, {0xc00d6ec000?, 0x83b211?, 0xc00d6ec005?})
        	GOROOT/src/net/fd_posix.go:68 +0x25
        net.(*conn).Read(0xc00e501668, {0xc00d6ec000?, 0x0?, 0x9ffb?})
        	GOROOT/src/net/net.go:196 +0x45
        github.com/cockroachdb/cockroach/pkg/util/cidr.metricsConn.Read({{0x7e82220?, 0xc00e501668?}, 0xc002919b20?, 0xc002919b30?}, {0xc00d6ec000?, 0x430ff4?, 0xc00d6ec005?})
        	pkg/util/cidr/cidr.go:519 +0x3e
        crypto/tls.(*atLeastReader).Read(0xc00bc44318, {0xc00d6ec000?, 0x9f0a?, 0x4b859d?})
        	GOROOT/src/crypto/tls/conn.go:816 +0x3b
        bytes.(*Buffer).ReadFrom(0xc00b8d1428, {0x7ddb100, 0xc00bc44318})
        	GOROOT/src/bytes/buffer.go:217 +0x98
        crypto/tls.(*Conn).readFromUntil(0xc00b8d1188, {0x7f8c5367c3a8, 0xc00d723a20}, 0x45b394?)
        	GOROOT/src/crypto/tls/conn.go:838 +0xde
        crypto/tls.(*Conn).readRecordOrCCS(0xc00b8d1188, 0x0)
        	GOROOT/src/crypto/tls/conn.go:627 +0x3db
        crypto/tls.(*Conn).readRecord(...)
        	GOROOT/src/crypto/tls/conn.go:589
        crypto/tls.(*Conn).Read(0xc00b8d1188, {0xc00d4ab000, 0x1000, 0xc00e53db00?})
        	GOROOT/src/crypto/tls/conn.go:1392 +0x145
        net/http.(*persistConn).Read(0xc00d31eb40, {0xc00d4ab000?, 0x91a725?, 0x7dcc5e0?})
        	GOROOT/src/net/http/transport.go:2125 +0x47
        bufio.(*Reader).fill(0xc00d49f320)
        	GOROOT/src/bufio/bufio.go:113 +0x103
        bufio.(*Reader).Peek(0xc00d49f320, 0x1)
        	GOROOT/src/bufio/bufio.go:152 +0x53
        net/http.(*persistConn).readLoop(0xc00d31eb40)
        	GOROOT/src/net/http/transport.go:2278 +0x172
        created by net/http.(*Transport).dialConn in goroutine 7279
        	GOROOT/src/net/http/transport.go:1947 +0x174f
        goroutine 7229 [select, "job":"1156248981336588289", "n":"1", "restore-span":"1", "restore-worker":"1"]:
        net/http.(*persistConn).writeLoop(0xc00d67d0e0)
        	GOROOT/src/net/http/transport.go:2600 +0xe7
        created by net/http.(*Transport).dialConn in goroutine 7304
        	GOROOT/src/net/http/transport.go:1948 +0x17a5
        
        goroutine dump: /artifacts/tmp/_tmp/b565fc69e118137eccfc1c549c30035c/goroutine_dump_56719054.txt
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_legacy/no_kms (53.37s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21253150?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/21253150?buildTab=artifacts#/) on master @ [74eff753539d441da4247b99f3fd408a3708fe1e](https://github.com/cockroachdb/cockroach/commits/74eff753539d441da4247b99f3fd408a3708fe1e):


```
=== RUN   TestCloudBackupRestoreAzure
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/b565fc69e118137eccfc1c549c30035c/logTestCloudBackupRestoreAzure2270805893
    test_log_scope.go:82: use -show-logs to present logs inline
    backup_cloud_test.go:268: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b565fc69e118137eccfc1c549c30035c/logTestCloudBackupRestoreAzure2270805893
--- FAIL: TestCloudBackupRestoreAzure (91.97s)
=== RUN   TestCloudBackupRestoreAzure/storage_auth_explicit/kms_auth_explicit
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    backup_test.go:1002: error scanning '&{0xc0104f2ea0 <nil>}': pq: experimental online restore: encryption not supported
    testutils.go:368: no Invalid Descriptors
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_explicit/kms_auth_explicit (12.21s)
=== RUN   TestCloudBackupRestoreAzure/storage_auth_explicit/kms_auth_implicit
    backup_test.go:1002: error scanning '&{0xc00d59ab40 <nil>}': pq: experimental online restore: encryption not supported
    testutils.go:368: no Invalid Descriptors
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_explicit/kms_auth_implicit (11.37s)
=== RUN   TestCloudBackupRestoreAzure/storage_auth_implicit/kms_auth_explicit
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    backup_test.go:1002: error scanning '&{0xc01a8a0ea0 <nil>}': pq: experimental online restore: encryption not supported
    testutils.go:368: no Invalid Descriptors
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_implicit/kms_auth_explicit (11.45s)
=== RUN   TestCloudBackupRestoreAzure/storage_auth_implicit/kms_auth_implicit
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    backup_test.go:1002: error scanning '&{0xc015000b40 <nil>}': pq: experimental online restore: encryption not supported
    testutils.go:368: no Invalid Descriptors
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_implicit/kms_auth_implicit (11.11s)
=== RUN   TestCloudBackupRestoreAzure/storage_auth_legacy/kms_auth_explicit
    backup_test.go:1002: error scanning '&{0xc011799200 <nil>}': pq: experimental online restore: encryption not supported
    testutils.go:368: no Invalid Descriptors
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_legacy/kms_auth_explicit (13.61s)
=== RUN   TestCloudBackupRestoreAzure/storage_auth_legacy/kms_auth_implicit
    backup_test.go:1002: error scanning '&{0xc015580b40 <nil>}': pq: experimental online restore: encryption not supported
    testutils.go:368: no Invalid Descriptors
--- FAIL: TestCloudBackupRestoreAzure/storage_auth_legacy/kms_auth_implicit (12.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/backup/backup_cloud_test.go
+++ b/pkg/backup/backup_cloud_test.go
@@ -14,6 +14,7 @@ import (
 
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/cockroachdb/cockroach/pkg/backup/backuptestutils"
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/cloud"
 	"github.com/cockroachdb/cockroach/pkg/cloud/amazon"
@@ -171,6 +172,9 @@ func TestCloudBackupRestoreGoogleCloudStorage(t *testing.T) {
 func TestCloudBackupRestoreAzure(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	// Fast restore does not yet support encrypted backups.
+	backuptestutils.DisableFastRestoreForTest(t)
+
 	accountName := os.Getenv("AZURE_ACCOUNT_NAME")
 	if accountName == "" {
 		skip.IgnoreLint(t, "AZURE_ACCOUNT_NAME env var must be set")
@@ -272,6 +276,8 @@ func TestCloudBackupRestoreAzure(t *testing.T) {
 func TestCloudBackupRestoreKMSInaccessibleMetric(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	// Fast restore does not yet support encrypted backups.
+	backuptestutils.DisableFastRestoreForTest(t)
 
 	azureVaultName := os.Getenv("AZURE_VAULT_NAME")
 	if azureVaultName == "" {

